cosmpy/aerial/tx.py

```diff
@@ -41,6 +42,163 @@
+    )
+    tx.sign(sender.signer(), client.network_config.chain_id, account.number)
+    tx.complete()
+
+    # simulate the gas and fee for the transaction
+    gas_limit, fee = client.estimate_gas_and_fee_for_tx(tx)
+
+    return gas_limit, fee
+
+
+class TxFee:
+    """Cosmos SDK TxFee abstraction."""
+
+    def __init__(
+        self,
+        amount: List["Coin"],  # type: ignore # noqa: F821
+        gas_limit: int,
+        granter: Optional[Address] = None,
+        payer: Optional[Address] = None,
+    ):
+        """Init the Transaction fee class.
+
+        :param amount: tx fee amount
+        :param gas_limit: gas limit
+        :param granter: transaction fee granter, defaults to None
+        :param payer: transaction fee payer, defaults to None
+        :returns initialised TxFee
+        """
+        self.amount = amount
+        self.gas_limit = gas_limit
+        self.granter = granter
+        self.payer = payer
+
+    @classmethod
+    def from_fixed_amount(
+        cls,
+        amount: str,
+        gas_limit: int = 0,
+        granter: Optional[Address] = None,
+        payer: Optional[Address] = None,
+    ) -> "TxFee":
+        """Init the Transaction fee class from string amount.
+
+        :param amount: tx fee amount string
+        :param gas_limit: gas limit
+        :param granter: transaction fee granter, defaults to None
+        :param payer: transaction fee payer, defaults to None
+        :return: TxFee
+        """
+        return cls(
+            amount=parse_coins(amount),
+            gas_limit=gas_limit,
+            granter=granter,
+            payer=payer,
+        )
+
+    @classmethod
+    def from_gas_only(
+        cls,
+        client: "LedgerClient",  # type: ignore # noqa: F821
+        gas_limit: int,
+        granter: Optional[Address] = None,
+        payer: Optional[Address] = None,
+    ) -> "TxFee":
+        """Init the Transaction fee class from gas limit.
+
+        :param client: Ledger client
+        :param gas_limit: gas limit
+        :param granter: transaction fee granter, defaults to None
+        :param payer: transaction fee payer, defaults to None
+        :return: TxFee
+        """
+        estimated_amount = client.estimate_fee_from_gas(gas_limit)
+        return cls(
+            amount=parse_coins(estimated_amount),
+            gas_limit=gas_limit,
+            granter=granter,
+            payer=payer,
+        )
+
+    @classmethod
+    def from_simulation(
+        cls,
+        client: "LedgerClient",  # type: ignore # noqa: F821
+        tx: "Transaction",  # type: ignore  # noqa: F821
+        sender: "Wallet",  # type: ignore # noqa: F821
+        amount: Optional[str] = None,
+        granter: Optional[Address] = None,
+        payer: Optional[Address] = None,
+        account: Optional["Account"] = None,  # type: ignore # noqa: F821
+        memo: Optional[str] = None,
+    ) -> Tuple["TxFee", Optional["Account"]]:  # type: ignore # noqa: F821
```

Comment:
I would add the `Tx` instance to the returned tuple = the return value would contain modified & sealed `Tx` instance with filled in `Fee` instance from simulation.